tests/search/test_Route.py

```diff
@@ -322,23 +322,29 @@ def test_str_contains_route(ok_small, locs: list[int]):
 
 def test_route_duration_access(ok_small):
     """
-    Tests access to a client's or depot's duration segment, as tracked by
-    the route.
+    Tests access to a client's or depot's duration segment, as tracked by the
+    route. The start depot's segment differs from the end depot's segment due
+    to different latest start and latest finish on the vehicle type.
     """
+    vehicle_type = ok_small.vehicle_type(0).replace(start_late=10_000)
+    ok_small = ok_small.replace(vehicle_types=[vehicle_type])
```

Comment:
Is this test no longer valid without changes? I think it should still work? If so, can we not modify it, and instead only add new tests like the one below?

Reply:
The test is still valid without changes. However, personally I prefer extending/modifying an existing test over adding an entire new test which is very similar to the existing test. But It is fine for me to leave the current test and create a new one for the new logic.